osu.Game/Screens/Ranking/SoloResultsScreen.cs

```diff
@@ -1,90 +1,130 @@
 
         [Resolved]
-        private RulesetStore rulesets { get; set; } = null!;
-
-        [Resolved]
-        private IAPIProvider api { get; set; } = null!;
+        private LeaderboardManager leaderboardManager { get; set; } = null!;
 
         public SoloResultsScreen(ScoreInfo score)
             : base(score)
         {
         }
 
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            globalScores.BindTo(leaderboardManager.Scores);
+        }
+
         protected override async Task<ScoreInfo[]> FetchScores()
         {
             Debug.Assert(Score != null);
 
             if (Score.BeatmapInfo!.OnlineID <= 0 || Score.BeatmapInfo.Status <= BeatmapOnlineStatus.Pending)
                 return [];
 
-            var requestTaskSource = new TaskCompletionSource<APIScoresCollection>();
+            var criteria = new LeaderboardCriteria(
+                Score.BeatmapInfo!,
+                Score.Ruleset,
+                leaderboardManager.CurrentCriteria?.Scope ?? BeatmapLeaderboardScope.Global,
+                leaderboardManager.CurrentCriteria?.ExactMods
+            );
+            var requestTaskSource = new TaskCompletionSource<LeaderboardScores>();
+            globalScores.BindValueChanged(_ =>
+            {
+                if (globalScores.Value != null && leaderboardManager.CurrentCriteria?.Equals(criteria) == true)
+                    requestTaskSource.TrySetResult(globalScores.Value);
+            });
+            leaderboardManager.FetchWithCriteria(criteria, forceRefresh: true);
 
-            getScoreRequest = new GetScoresRequest(Score.BeatmapInfo, Score.Ruleset);
-            getScoreRequest.Success += requestTaskSource.SetResult;
-            getScoreRequest.Failure += requestTaskSource.SetException;
-            api.Queue(getScoreRequest);
+            var result = await requestTaskSource.Task.ConfigureAwait(false);
 
-            try
+            if (result.FailState != null)
             {
-                var scores = await requestTaskSource.Task.ConfigureAwait(false);
-                var toDisplay = new List<ScoreInfo>();
+                Logger.Log($"Failed to fetch scores (beatmap: {Score.BeatmapInfo}, ruleset: {Score.Ruleset}): {result.FailState}");
+                return [];
+            }
+
+            var clonedScores = result.AllScores.Select(s => s.DeepClone()).ToArray();
 
-                for (int i = 0; i < scores.Scores.Count; ++i)
+            List<ScoreInfo> sortedScores = [];
+
+            foreach (var clonedScore in clonedScores)
+            {
+                // ensure that we do not double up on the score being presented here.
+                // additionally, ensure that the reference that ends up in `sortedScores` is the `Score` reference specifically.
+                // this simplifies handling later.
+                if (clonedScore.Equals(Score) || clonedScore.MatchesOnlineID(Score))
```

Comment:
The added equality check here is necessary because of offline scores - previously, because all scores retrieved here were online, online ID matching was sufficient, but it no longer is after results can use local leaderboards.